content/security.md

```diff
@@ -0,0 +1,380 @@
+
+1. **Methods**: Any data that comes in through method arguments needs to be validated, and methods should not return data the user shouldn't have access to.
+2. **Publications**: Any data that comes in through publication arguments needs to be validated, and publications should not return data the user shouldn't have access to.
+3. **Served files**: You should make sure none of the source code or configuration files served to the client have secret data.
+
+Each of these points will have their own section below.
+
+#### Don't use Collection.allow/deny
+
+In this guide, we're going to take a strong position that using [allow](http://docs.meteor.com/#/full/allow) or [deny](http://docs.meteor.com/#/full/deny) to run MongoDB queries directly from the client is not a good idea. The main reason is that it is very hard to follow the principles outlined above. It's extremely hard to validate the complete space of possible MongoDB operators, which could potentially grow over time.
+
+There have been several articles about the potential pitfalls of accepting MongoDB update operators from the client, in particular the [Allow & Deny Security Challenge](https://www.discovermeteor.com/blog/allow-deny-security-challenge/) and its [results](https://www.discovermeteor.com/blog/allow-deny-challenge-results/), both on the Discover Meteor blog.
+
+Given the points above, we're going to recommend that all Meteor apps should use Methods to accept data input from the client, and restrict the arguments accepted by each Method as tightly as possible.
+
+Here's a code snippet to disable client-side updates on a collection; this will make sure no other part of the code can use `allow`:
+
+```js
+// Deny all client-side updates on the Lists collection
+Lists.deny({
+  insert() { return true },
+  update() { return true },
+  remove() { return true },
+});
+```
+
+
+### Methods
+
+Methods are the way your Meteor server accepts inputs and data from the outside world, so it's natural that they are the most important topic for security. If you don't properly secure your methods, you could result in users modifying your database in unexpected ways - editing other people's documents, deleting data, or messing up your database schema causing your app to crash.
+
+#### The mdg:method package
+
+To help you write good methods, we've written a simple wrapper package for Methods that enforces argument validation using `aldeed:simple-schema`. XXX elaborate here
+
+#### Always specify a schema for your method arguments
+
+This is trivial when using the `mdg:method` package, and you can also achieve the same effect by using [`check`](http://docs.meteor.com/#/full/check) inside a vanilla method. The idea is that you don't want someone to pass a data type you aren't expecting, which could mess up some of your logic.
```

Comment:
Rather than "mess up some of your logic", I'd refer to a real problem -- for instance, pass an object into a mongo query where you were expecting a string..

```
Meteor.methods({
  removeDoc: (id) => {
    X.remove(id);
  }
});
```